Public/Src/Cache/ContentStore/Distributed/NuCache/ClusterState.cs

```diff
@@ -162,5 +208,10 @@ public Result<MachineLocation> GetRandomMachineLocation(IReadOnlyList<MachineLoc
                 return new Result<MachineLocation>("Could not select a machine location.");
             }
         }
+
+        /// <summary>
+        /// Gets the set of locations designated for a hash. This is relevant for proactive copies and eviction.
+        /// </summary>
+        public MachineLocation[] GetDesignatedLocations(ContentHash hash) => _binManager.GetLocations(hash);

```

Comment:
Will fail with NRE if _binManager is null.

Does it make sense to add a NullBinManager? or just create it all the time?